CesiumAsync/include/CesiumAsync/Impl/WithTracing.h

```diff
@@ -9,7 +9,23 @@ namespace Impl {
 //! @cond Doxygen_Suppress
 
 template <typename Func, typename T> struct WithTracing {
-  static auto wrap([[maybe_unused]] const char* tracingName, Func&& f) {
+  static auto
+  begin([[maybe_unused]] const char* tracingName, [[maybe_unused]] Func&& f) {
+#if CESIUM_TRACING_ENABLED
+    return
+        [tracingName, CESIUM_TRACE_LAMBDA_CAPTURE_TRACK()](T&& result) mutable {
+          CESIUM_TRACE_USE_CAPTURED_TRACK();
+          if (tracingName) {
+            CESIUM_TRACE_BEGIN_IN_TRACK(tracingName);
+          }
+          return std::move(result);
+        };
+#else
+    return Impl::unwrapFuture<Func, T>(std::forward<Func>(f));
```

Comment:
I maybe wrong here, but I think the type `f` of this function is rvalue reference and not universal reference, so returning std::move(f) is appropriate in this case instead of std::forward

Reply:
Ah good catch! Though, we actually do want perfect forwarding here, so I tweaked the templates so that `f` will be a universal reference. I made a similar mistake in `unwrapFuture` so I fixed it there too.